Kitware/VolView
# Remote state

### New Features
- Loads a `session.volview.zip` state file if in `files` URL parameter.
- Fetch remotely loaded datasets from URLs in state file, rather than zipping in all files.

### How
- Introduce `DatasetFile` type with copyleft `{ url, remoteFilename }` properties.  
- Files downloaded from url parameters, JSON manifest files, and sample datasets get wrapped as a `DatasetFile`.  
- `datasets-dicom` and `datasets-images` eventualy pass `DatasetFile`s to `datasets-files`.
 - On Save State: DatasetFiles with `url` and `remoteFilename` are saved as "Remote Datasets" in the JSON manifest and their raw data is not zipped into the state file.  `state-file/utils.ts` does this and is effectively has `datasets-files`' serialize and deserialize functions.

### Test Plan

For each scenario, verify:
- Save session has no errors
- Remote datasets are not included in .zip
- Load session zip file has no errors and state is restored

Scenarios
- [ ] Multiple uncompressed remote DICOM files
- [ ] Multiple zipped remote DICOM files

Also:
- [ ] A session zip file is loaded when pointed to by URL files parameter
- [ ] Load files from JSON manifest, save session, verify files are not in zip, verify session zip file loads.
